### ISSUE TITLE
NullPointerExeception from FlatComboBoxUI

### ISSUE THREAD

**Issue by BadHorseMonkey** [JFormDesigner/FlatLaf#408]
Hi All,

Using FlatLaf, we sometimes see this error.  I can't tell if this is a FlatLaf issue, or if we have't initialized something.  If it's us, I wish I had a better error message that told me which component was incorrectly specified.  Unfortunately, this is 14 year old Open Source project and has a lot of widely dispersed older code.

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "javax.swing.JComponent.setBorder(javax.swing.border.Border)" because "this.rendererComponent" is null
	at com.formdev.flatlaf.ui.FlatComboBoxUI$CellPaddingBorder.install(FlatComboBoxUI.java:822)
	at com.formdev.flatlaf.ui.FlatComboBoxUI.getSizeForComponent(FlatComboBoxUI.java:571)
	at java.desktop/javax.swing.plaf.basic.BasicComboBoxUI.getDefaultSize(BasicComboBoxUI.java:1395)
	at com.formdev.flatlaf.ui.FlatComboBoxUI.getDefaultSize(FlatComboBoxUI.java:542)
	at java.desktop/javax.swing.plaf.basic.BasicComboBoxUI.getDisplaySize(BasicComboBoxUI.java:1466)
	at com.formdev.flatlaf.ui.FlatComboBoxUI.getDisplaySize(FlatComboBoxUI.java:550)
	at java.desktop/javax.swing.plaf.basic.BasicComboBoxUI.getMinimumSize(BasicComboBoxUI.java:999)
	at com.formdev.flatlaf.ui.FlatComboBoxUI.getMinimumSize(FlatComboBoxUI.java:533)
	at java.desktop/javax.swing.plaf.basic.BasicComboBoxUI.getPreferredSize(BasicComboBoxUI.java:988)
	at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1680)
	at java.desktop/java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:723)
	at java.desktop/java.awt.Container.preferredSize(Container.java:1824)
	at java.desktop/java.awt.Container.getPreferredSize(Container.java:1808)
	at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1682)
	at java.desktop/java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:733)
	at java.desktop/java.awt.Container.preferredSize(Container.java:1824)
	at java.desktop/java.awt.Container.getPreferredSize(Container.java:1808)
	at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1682)
	at java.desktop/java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:723)
	at java.desktop/java.awt.Container.preferredSize(Container.java:1824)
	at java.desktop/java.awt.Container.getPreferredSize(Container.java:1808)
	at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1682)
	at java.desktop/java.awt.CardLayout.preferredLayoutSize(CardLayout.java:280)
	at java.desktop/java.awt.Container.preferredSize(Container.java:1824)
	at java.desktop/java.awt.Container.getPreferredSize(Container.java:1808)
	at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1682)
	at java.desktop/java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:723)
	at java.desktop/java.awt.Container.preferredSize(Container.java:1824)
	at java.desktop/java.awt.Container.getPreferredSize(Container.java:1808)
	at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1682)
	at java.desktop/java.awt.BorderLayout.layoutContainer(BorderLayout.java:819)
	at java.desktop/java.awt.Container.layout(Container.java:1538)
	at java.desktop/java.awt.Container.doLayout(Container.java:1527)
	at java.desktop/java.awt.Container.validateTree(Container.java:1723)
	at java.desktop/java.awt.Container.validateTree(Container.java:1732)
	at java.desktop/java.awt.Container.validateTree(Container.java:1732)
	at java.desktop/java.awt.Container.validateTree(Container.java:1732)
	at java.desktop/java.awt.Container.validateTree(Container.java:1732)
	at java.desktop/java.awt.Container.validate(Container.java:1658)
	at java.desktop/javax.swing.RepaintManager$3.run(RepaintManager.java:745)
	at java.desktop/javax.swing.RepaintManager$3.run(RepaintManager.java:743)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/javax.swing.RepaintManager.validateInvalidComponents(RepaintManager.java:742)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1883)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by DevCharly** [comment]
Thanks for reporting. This is strange 😉 

This is the related code:

https://github.com/JFormDesigner/FlatLaf/blob/a253b6c0cf10c831a013acbbfbe94153c26c302f/flatlaf-core/src/main/java/com/formdev/flatlaf/ui/FlatComboBoxUI.java#L817-L822

The NullPointerException occurs in line 822 because field `rendererComponent` is `null`.
The strage thing is that the local variable `jc`, which is **not null**, is assigned to `rendererComponent` earlier in line 818.
And there is not much code between lines 818 and 822 that could change `rendererComponent` to `null` 😕 
So either there is some threading involved that access combobox renderer.
Or there is some code in `getBorder()` (invoked at line 821) that access combobox renderer...

Anyway the fix in FlatLaf is easy.
I'll change `rendererComponent.setBorder( this );` to `jc.setBorder( this );`...

**Comment by DevCharly** [comment]
Fixed in main branch.

Please try the latest snapshot:
https://github.com/JFormDesigner/FlatLaf#snapshots

**Comment by BadHorseMonkey** [comment]
thanks for the quick reply!  I've got the snapshot and I'll report back if I see anything.  This was a rare error, so it may be a while before I know anything.

**Comment by BadHorseMonkey** [comment]
Have tested the snapshot and it seems to be fixed.  Will wait for 2.0 before it goes in our prod code. :)

**Comment by DevCharly** [comment]
Thanks for the update.

**Comment by DevCharly** [comment]
Fixed in [1.6.2](https://github.com/JFormDesigner/FlatLaf/releases/tag/1.6.2).
